internal/irc/handler_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 - 2025, Ludvig Lundgren and the autobrr contributors.
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package irc
+
+import (
+	"testing"
+
+	"github.com/autobrr/autobrr/internal/domain"
+	"github.com/autobrr/autobrr/internal/notification"
+	"github.com/autobrr/autobrr/internal/release"
+	"github.com/r3labs/sse/v2"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInitAndRemoveIndexers(t *testing.T) {
+	def1 := &domain.IndexerDefinition{
+		Identifier: "abc",
+		IRC: &domain.IndexerIRC{
+			Channels:   []string{"1", "2"},
+			Announcers: []string{"3", "4"},
+			Parse: &domain.IndexerIRCParse{
+				Lines: []domain.IndexerIRCParseLine{},
+			},
+		},
+	}
+	def2 := &domain.IndexerDefinition{
+		Identifier: "def",
+		IRC: &domain.IndexerIRC{
+			Channels:   []string{"5", "6"},
+			Announcers: []string{"7", "8"},
+			Parse: &domain.IndexerIRCParse{
+				Lines: []domain.IndexerIRCParseLine{},
+			},
+		},
+	}
+	definitions := []*domain.IndexerDefinition{def1, def2}
+
+	log := zerolog.Logger{}
+	sse := new(sse.Server)
+	network := domain.IrcNetwork{}
+	releaseSvc := (release.Service)(nil)
+	notificationSvc := (notification.Service)(nil)
```

Comment:
Not a huge fan of passing in typed nils, but I wanted to use the `NewHandler` code to make sure the setup/teardown is consistent with what would actually be executed in the running code.